resources/turso/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+use async_trait::async_trait;
+use libsql_client::{client::Client, new_client_from_config, Config};
+use serde::{Deserialize, Serialize};
+use shuttle_service::{
+    error::{CustomError, Error as ShuttleError},
+    Factory, ResourceBuilder, Type,
+};
+use url::Url;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct Turso {
+    addr: String,
+    token: String,
+    local_addr: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct TursoOutput {
+    conn_url: Url,
+    token: Option<String>,
+}
+
+impl Turso {
+    pub fn local_addr(mut self, local_addr: &str) -> Self {
+        self.local_addr = Some(local_addr.to_string());
+        self
+    }
+}
+
+pub enum Error {
+    UrlParseError(url::ParseError),
+}
+
+impl From<Error> for shuttle_service::Error {
+    fn from(error: Error) -> Self {
+        let msg = match error {
+            Error::UrlParseError(err) => format!("Cannot parse Turso Url: {}", err),
+        };
+
+        ShuttleError::Custom(CustomError::msg(msg))
+    }
+}
+
+#[async_trait]
+impl ResourceBuilder<Client> for Turso {
+    const TYPE: Type = Type::Turso;
+
+    type Config = Self;
+    type Output = TursoOutput;
+
+    fn new() -> Self {
+        Self::default()
+    }
+
+    fn config(&self) -> &Self::Config {
+        &self
+    }
+
+    async fn output(
+        self,
+        factory: &mut dyn Factory,
+    ) -> Result<Self::Output, shuttle_service::Error> {
+        match factory.get_environment() {
+            shuttle_service::Environment::Production => {
+                if self.addr.is_empty() {
+                    // XXX: should we raise an error even not running in a production
+                    // environment ?
```

Comment:
Not sure I understand this. If we are in this match arm, we are definitely running in production, are we not?

Reply:
The question was that if we run in a local environment, should we raise an error ? The idea behind it is to be able to warn even when running locally that you're going to have an issue when deploying.

The other integrations don't seem to do that though, and maybe a warning message is enough.

Reply:
Well, right now, if the `addr` is `None`, we fall back to a local database file, which is the intended way of running locally anyway, right?